### PR TITLE
Add JEDI QG model example cycled DA workflow

### LINKED ISSUE #84: Need test JEDI workflow
The Chiltepin repository needs an example Parsl workflow to use for testing and demonstration purposes.  The workflow should use JEDI applications, be simple enough to complete and run in a short period of time, and be complex enough to demonstrate basic DA and cycling features.

### PR DESCRIPTION
**Pull request by christopherwharrop-noaa**
This PR adds a collection of Parsl wrappers that configure and execute the various applications needed for a simple cycled 3DVar JEDI QG workflow.  The wrappers consist of "configure" and "run" tasks that execute separately on different resources.  The "configure" tasks will run on either a "service" (when internet access is required) or a "serial" resource while the "run" tasks can execute on compute nodes in parallel.  Wrappers are created for: installing JEDI, running a "truth" forecast, creating simulated observations from the "truth", running 3DVar using the simulated obs and a background from a previous cycle forecast, and running a forecast initialized from the analysis produced by 3DVar.

These are not the most efficient or best organized wrappers.  Much more thought should be put into design that can be applicable for future applications and experiment types.  However, effort was made to build configurable components that could be used to run cycled 3DVar experiments with varying configuration options.

An experiment program, `workflow.py` was created to illustrate how the wrappers can be used to build a workflow.

Automated tests for the various wrappers has not been added yet because it isn't clear what the best way to do it is.  All such tests would depend on a JEDI install, which takes time to build, and would take even longer in CI.  Testing will be added once a strategy is determined.

Closes #84

### CODE REVIEW

File: src/chiltepin/jedi/leadtime.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+def leadtime_to_seconds(fcst):
```
Comment (NaureenBharwaniNOAA) [inline src/chiltepin/jedi/leadtime.py:1]: if we renamed this from `fcst_to_seconds` to `leadtime_to_seconds`. Should we also take advantage of this change and fix the parameter being passed in from `fcst` to `leadtime`?

Comment (christopherwharrop-noaa) [reply src/chiltepin/jedi/leadtime.py:1]: That's a great idea!  It does look confusing now.

File: tests/test_leadtime.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    assert leadtime_to_seconds("MT0S") == -0
+    assert leadtime_to_seconds("MT1H") == -3600
+    assert leadtime_to_seconds("MT1H30M") == -3600
+
+    assert leadtime_to_seconds("M1D") == -86400
+    assert leadtime_to_seconds("M1DT1H") == -90000
+    assert leadtime_to_seconds("M1DT1H30M") == -90000
+    assert leadtime_to_seconds("M1DT1H30M") == -90000
+    
+
```
Comment (NaureenBharwaniNOAA) [inline tests/test_leadtime.py:22]: These numbers are pretty off, also there are only 13 entries for `leadtime_to_seconds` and 14 supplied values for `seconds_to_leadtime`. So the values are a little off. The numbers in the code accurately pass the tests. 

`leadtime.leadtime_to_seconds`

- "PT0S"
- **_"PT1M"_**
- "PT1H"
- **_"PT1H30M"_**

- "P1D"
- "P1DT1H"
- **_"P1DT1H30M"_**

- "MT0S"
- "MT1H"
- **_"MT1H30M"_**

- "M1D"
- "M1DT1H"
- **_"M1DT1H30M"_**


`leadtime.seconds_to_leadtime`
- 0
- 60
- 3600
- 5400

- 86400
- 90000
- 91800

- -0
- **_-60_**
- -3600
- -5400

- -86400
- -90000
- -91800

File: src/chiltepin/jedi/leadtime.py
Comment (NaureenBharwaniNOAA) [inline src/chiltepin/jedi/leadtime.py:1]: Changed all instances of `fcst` to `leadtime`

File: tests/test_leadtime.py (hunk 2 above)
Comment (christopherwharrop-noaa) [reply tests/test_leadtime.py]: `assert leadtime_to_seconds("PT1M") == 0`
should be
`assert leadtime_to_seconds("PT1M") == 60`

Comment (christopherwharrop-noaa) [reply tests/test_leadtime.py]: `assert leadtime_to_seconds("PT1H30M") == 3600`
should be
`assert leadtime_to_seconds("PT1H30M") == 5400`

Comment (christopherwharrop-noaa) [reply tests/test_leadtime.py]: `assert leadtime_to_seconds("P1DT1H30M") == 90000`
should be
`assert leadtime_to_seconds("P1DT1H30M") == 91800`

Comment (christopherwharrop-noaa) [reply tests/test_leadtime.py]: `assert leadtime_to_seconds("MT1H30M") == -3600`
should be
`assert leadtime_to_seconds("MT1H30M") == -5400`

Comment (christopherwharrop-noaa) [reply tests/test_leadtime.py]: `assert leadtime_to_seconds("M1DT1H30M") == -90000`
should be
`assert leadtime_to_seconds("M1DT1H30M") == -91800`

Comment (christopherwharrop-noaa) [reply tests/test_leadtime.py]: It appears I accidentally left out the `MT1M` case.  That should evaluate to `-60`.  Perhaps the missing case caused mapping from leadtime to seconds to get messed up in the issue.  Sorry about that.  I'll go edit the issue and fix it.

**Pull request closed without merging**